src/MongoDB.EntityFrameworkCore/Serializers/CollectionSerializationProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Serializers;
+
+namespace MongoDB.EntityFrameworkCore.Serializers;
+
+internal class CollectionsSerializationProvider : BsonSerializationProviderBase
+{
+    /// <inheritdoc/>
+    public override IBsonSerializer GetSerializer(Type type, IBsonSerializerRegistry serializerRegistry)
+    {
+        ArgumentNullException.ThrowIfNull(type);
+
+        if (type.IsArray)
+        {
+            return CreateArraySerializer(type, type.GetElementType()!, serializerRegistry);
+        }
+
+        if (type is {IsGenericType: true, ContainsGenericParameters: true})
+        {
+            throw new ArgumentException($"Generic type '{type.ShortDisplayName()}' has unassigned generic type parameters.", nameof(type));
+        }
+
+        return CreateCollectionSerializer(type, serializerRegistry)
+               ?? throw new ArgumentException($"No known serializer for type '{type.ShortDisplayName()}'.", nameof(type));
+    }
+
+
+    private IBsonSerializer CreateArraySerializer(Type type, Type elementType, IBsonSerializerRegistry serializerRegistry)
+        => type.GetArrayRank() switch // Only single dimensional arrays are currently supported in EF Core
```

Comment:
Is this comment valid?

Reply:
Will revisit multi-dimensional arrays and dictionaries in future post-preview-2 updates.

Reply:
Will remove all non-single dimension array code. It's possible it works now but it's out of scope and low-priority compared to other items on the backlog.